playbooks/roles/openstack-remove-compute/tasks/main.yml

```diff
@@ -0,0 +1,47 @@
+---
+- name: Get openstack compute node id
+  shell: |
+     set -o pipefail
+     openstack compute service list | grep {{ compute_node_name }} | awk '{print $2}'
+  register: compute_id
+  delegate_to: localhost
+  failed_when: false
+
+- name: Validate the compute node id
+  fail:
+    msg: "Compute node {{ compute_node_name }} does not exists in openstack"
+  delegate_to: localhost
+  when: compute_id.stdout == ""
+
+- name: Remove label from compute node
+  command: "kubectl label node {{ compute_node_name }} {{ item }}-"
+  with_items:
+    - openstack-compute-node
+    - openvswitch
+  delegate_to: localhost
+
+- name: Wait for compute pods to terminate
+  pause:
+    minutes: 3
+  delegate_to: localhost
+
+- name: Delete openstack compute service
+  command: "openstack compute service delete {{compute_id.stdout}}"
+  delegate_to: localhost
+
+- name: Check for control-plane label.
+  shell: |
+     set -o pipefail
+     kubectl get node {{ compute_node_name }} --show-labels | grep openstack-control-plane=enabled
+  register: node_label_result
+  delegate_to: localhost
+  failed_when: false
+
+- name: Delete subvolumes from compute node.
+  import_role:
+    name: airship-configure-worker
+    tasks_from: subvolumes-setup.yml
```

Comment:
> I agree that if the node is also used as openstack control, you don't want to delete the neutron subvolume, but we should delete the other subvolumes.

But we currently reboot after cleanup. so cleaning up only if they are compute node.

Reply:
sure, that's fine if we have to reboot the node. maybe a warning in the user guide? It's not a common produciton use case, more a dev/test env, maybe?

Reply:
> sure, that's fine if we have to reboot the node. maybe a warning in the user guide? It's not a common produciton use case, more a dev/test env, maybe?

Currently I am not rebooting the node because it can be controller node too. But deleting subvolumes and unmount directories except those are used by controller.